tests/wildcards.rs

```diff
@@ -0,0 +1,26 @@
+
+use std::process::Command;
+use std::fs::read_dir;
+use std::process::Stdio;
+
+#[test]
+fn download_only_wildcard_filtered() {
+    let download_path = "./website";
+    let mut cmd = Command::new("./target/debug/suckit")
+    .args(&["http://www.hochmuth.com/mp3-samples.htm", 
+            "-o", download_path,
+            "-w", "mp[3-4]",
+            "-j", "16"])
+    .stdout(Stdio::inherit())
+    .stderr(Stdio::inherit())
+    .spawn()
+    .unwrap();
+    
+    let status = cmd.wait();
+    println!("Exited with status {:?}", status);
+
+    let paths = read_dir("./website/").unwrap();
+
+    assert_eq!(paths.count(), 9);
```

Comment:
I think the test is good, but if the page changes it will suddenly become invalid. Also, I'm not sure it shows the wildcards are working since the website only contains .mp3 samples. It's a good test nonetheless, but I suggest changing the `assert_eq!(paths.count(), 9)` with the output of a command that counts .mp3 files, for example `echo *.mp3 | wc -w` assuming all filenames are single words.

Reply:
> I think the test is good, but if the page changes it will suddenly become invalid. 

Yes maybe a server on localhost on the future.

> Also, I'm not sure it shows the wildcards are working since the website only contains .mp3

There is also html files

Reply:
> Also, I'm not sure it shows the wildcards are working since the website only contains .mp3 samples

Yes, it does work. I will improve the tests to make this explicit.
> Yes maybe a server on localhost on the future.

Yep, I am putting together a simple webserver fixture seving a sample index.html with different files for the tests.